rakelib/docker_debugging.rake

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+$stdout.sync = true
+namespace :docker_debugging do
+  desc 'Setup environment for debugging in docker'
+  task setup: :environment do |_task|
+    if Settings.docker_debugging
+      raise 'This rake task runs in development mode only!' unless Rails.env.development?
+
+      s = Settings.docker_debugging
+      if s.hang_around
+        puts 'Just hanging around, you can:'
+        puts 'docker exec -it some_container_id bash'
+        Thread.send(:new) { loop {} }.join
+      end
+      if s.set_up_db
+        puts 'Starting db setup'
+        Rake::Task['db:drop'].invoke
+        Rake::Task['db:create'].invoke
+        Rake::Task['db:schema:load'].invoke
+        Rails.env = 'test'
+        Rake::Task['db:schema:load'].invoke
+        puts 'All dun!'
+      end
+      sh 'foreman start -m all=1,clamd=0,freshclam=0'
+    end
+  end
+end
```

Comment:
Why do we need a new rake task for this? What are we trying to solve?

Reply:
From docker-entrypoint.sh:
```
if [ -e  "./docker_debugging" ] ; then
  echo starting rake docker_debugging:setup
  rake docker_debugging:setup
fi
```

After a conversation with Bastos he gave the setup he uses to set up the DB.  
From the local settings (above) I drive the behaviour docker-compose up does for me.

It sets up my DB (if that flag is set) then starts the rails server.  Usually, the DB flag is false so between 'ups' I persist my data.  There is also the 'hang_around' flag.  This ensures that vets_api container doesn't die allowing us to exec in and do whatever we want before any of the various services are started (it is our responsibility to start the rails server if desired).